modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/transaction/TransactionService.scala

```diff
@@ -0,0 +1,16 @@
+package org.tessellation.dag.l1.domain.transaction
+
+import cats.data.NonEmptyList
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{Transaction, TransactionReference}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+trait TransactionService[F[_]] {
+  def isParentAccepted(tx: Transaction): F[Boolean]
+  def getLastAcceptedReference(address: Address): F[TransactionReference]
+  def accept(hashedTx: Hashed[Transaction]): F[Unit]
+  def put(transactions: Set[Signed[Transaction]]): F[Unit]
+  def pull(): F[Option[NonEmptyList[Signed[Transaction]]]]
```

Comment:
Wouldn't it be better to just have just `F[List[Signed[Transaction]]]`?

Reply:
Don't know. I kind of like the explicit indication that we either get `None`, or `Some` transactions. The client side can default to an empty list if there is a need to have an empty transaction list.

Reply:
Regarding the `.gitkeep` file I think I did remove it 🤔 (at least GitHub shows I did)